numba/_dynfunc.c

```diff
@@ -462,15 +448,11 @@ static PyTypeObject GeneratorType = {
     0,                                        /* tp_del */
     0,                                        /* tp_version_tag */
     0,                                        /* tp_finalize */
-#if PY_MAJOR_VERSION == 3
-/* Python 3.8 has two slots, 3.9 has one. */
-#if PY_MINOR_VERSION > 7
     0,                         /* tp_vectorcall */
-#if PY_MINOR_VERSION == 8
+/* Python 3.8 has two slots, 3.9 has one. */
```

Comment:
Should this comment also indicate that everything beyond 3.9 also only has a single slot?

Reply:
I’m not sure how much value the comment adds here beyond what is described by the actual code. If the comment is maintained it could also be shortened to just “Python 3.8 has two slots” and make no mention of other specific python versions.

Reply:
@jamesobutler yes, shortening would also be an option. I am OK with that.

Reply:
I've pushed an additional commit to shorten the comment. I've rebased to let automated testing run against latest `main` changes.